pypsa/examples.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+This module contains functions for retrieving/loading example networks provided
+by the PyPSA project.
+"""
+
+import logging
+from urllib.request import urlretrieve as _retrieve
+import os
+
+from pathlib import Path
+from .components import Network
+
+logger = logging.getLogger(__name__)
+
+
+# for the writable data directory follow the XDG guidelines
+# https://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html
+_writable_dir = os.path.join(os.path.expanduser("~"), ".local", "share")
+_data_dir = os.path.join(
+    os.environ.get("XDG_DATA_HOME", os.environ.get("APPDATA", _writable_dir)),
+    "pypsa-examples",
+)
+_data_dir = Path(_data_dir)
+_data_dir.mkdir(exist_ok=True)
+
+
+# TODO: replace ending by 'master/' as soon as merging into master
+_repo_url = "https://github.com/PyPSA/PyPSA/raw/example-module/"
```

Comment:
Even better would be the version that is currently installed, right?

```suggestion
_repo_url = f"https://github.com/PyPSA/PyPSA/raw/v{pypsa.__version__}/"
```

Reply:
on the one hand this is more consistent, on the other you loose the availability to directly retrieve from the master branch. Perhaps a new flag `from_master` could provide a remedy?

Reply:
A `from_master=False` flag sounds good!